File: Recursion/lecture 6/Fraz_recursion_6.py
def reverseArrayHelper(nums, l, r):
    if(l >= r): return
    nums[r], nums[l] = nums[l], nums[r]
    return reverseArrayHelper(nums, l+1, r-1)

def reverseArray(nums):
    return reverseArrayHelper(nums, 0, len(nums)-1)

nums = list(map(int, input("Enter space seperated values : ").split()))
reverseArray(nums)
print(nums)

File: Recursion/lecture 6/Fraz_recursion_6.cpp
void reverseArrayHelper(int l,int r,vector<int> &arr)
{
    if(l>=r) return;
    swap(arr[l],arr[r]);
    reverseArrayHelper(l+1,r-1,arr);
}

void reverseArray(vector<int> &arr,int m)
{
    reverseArrayHelper(m+1,arr.size()-1,arr);
}